app/models/concerns/monthly_equivalent_calculator.rb

```diff
@@ -17,6 +17,7 @@ def calculate_monthly_equivalent!(target_field:, collection:, date_method: :paym
 
   def calculate_monthly_equivalent(collection:, date_method: :payment_date, amount_method: :amount)
     @converter = nil # reset the converter each time it's used
+    @frequency = nil
```

Comment:
Should we be memoizing `converter` and `frequency` at all?

Reply:
That can be one for the Eligibility Platform team to consider! 😄

Reply:
Converter has to be memoized due to the way the implementation is structured (the memoization is called once to do it, and again for the answer). The structure of this code in CFE-Civil has changed so that local variables are used - which is how we found the bug. We had an new test that accidentally relied on the bug, which then broke when we accidentally fixed it